enterprise_access/apps/bffs/handlers.py

```diff
@@ -69,29 +73,97 @@ def load_and_process(self):
         The method in this class simply calls common learner logic to ensure the context is set up.
         """
         try:
+            # Transform enterprise customer data
+            self.transform_enterprise_customers()
+
             # Retrieve and process subscription licenses. Handles activation and auto-apply logic.
-            # TODO: retrieve enterprise customer metadata,ENT-9629
-            # self.load_enterprise_customer()
             self.load_and_process_subscription_licenses()
 
             # Retrieve default enterprise courses and enroll in the redeemable ones
-            self.load_default_enterprise_courses()
-            self.enroll_in_redeemable_default_courses()
+            self.load_default_enterprise_enrollment_intentions()
+            self.enroll_in_redeemable_default_enterprise_enrollment_intentions()
         except Exception as e:  # pylint: disable=broad-exception-caught
             logger.exception("Error loading learner portal handler")
             self.add_error(
                 user_message="An error occurred while loading and processing common learner logic.",
                 developer_message=f"Error: {e}",
             )
 
+    def transform_enterprise_customers(self):
+        """
+        Transform enterprise customer metadata retrieved by self.context.
+        """
+        if enterprise_customer := self.context.data.get('enterprise_customer'):
+            self.context.data['enterprise_customer'] = self.transform_enterprise_customer(enterprise_customer)
+        if active_enterprise_customer := self.context.data.get('active_enterprise_customer'):
+            self.context.data['active_enterprise_customer'] =\
+                self.transform_enterprise_customer(active_enterprise_customer)
+        if enterprise_customer_users := self.context.data.get('all_linked_enterprise_customer_users'):
+            self.context.data['all_linked_enterprise_customer_users'] = [
+                self.transform_enterprise_customer_user(enterprise_customer_user)
+                for enterprise_customer_user in enterprise_customer_users
+            ]
+        if staff_enterprise_customer := self.context.data.get('staff_enterprise_customer'):
+            self.context.data['staff_enterprise_customer'] =\
+                self.transform_enterprise_customer(staff_enterprise_customer)
+
+    def transform_enterprise_customer_user(self, enterprise_customer_user):
+        """
+        Transform the enterprise customer user data.
+
+        Args:
+            enterprise_customer_user: The enterprise customer user data.
+        Returns:
+            The transformed enterprise customer user data.
+        """
+        enterprise_customer = enterprise_customer_user.get('enterprise_customer')
```

Comment:
The `enterprise_customer` dict in the ECU response payload won't be "full", right? Does that matter here?

Reply:
The `enterprise-learner` API uses the same `EnterpriseCustomerSerializer` that the list/detail enterprise customer API uses; it's the "full" metadata. This is the same enterprise customer metadata currently consumed by the frontends (both learner/admin).

That said, when cross-checking to verify, I noticed a discrepancy between `admin_users` list in the `enterprise-learner` API (this data source) vs. the `enterprise-customer` detail API. The `enterprise-learner` API _always_ denotes `admin_users` as an empty list, even if the customer has admin users...

I filed a [bug ticket](https://2u-internal.atlassian.net/browse/ENT-9741) and fixed the bug through this PR: https://github.com/openedx/edx-enterprise/pull/2287